spec/controllers/Api/v1/projects_controller_spec.rb

```diff
@@ -0,0 +1,124 @@
+RSpec.describe Api::V1::ProjectsController, type: :controller do
+  let(:user) { create(:user) }
+  let(:projects) { create_list(:project, 5, user: user) }
+  let(:project_id) { projects.last.id }
+
+  describe "GET /users/:user_id/projects" do
+    context "when user's projects exist" do
+      before { projects }
+      it "returns the user's projects" do
+        get :index, format: :json, params: { user_id: user.id }
+        expect(response.status).to eq(200)
+        users_response = JSON.parse(response.body)
+        expect(users_response).not_to be_empty
+        expect(users_response.size).to eq(5)
+      end
+    end
+
+    context "when user's projects do not exist" do
+      let(:projects) { [] }
+      it "returns an empty array with a 404" do
+        get :index, format: :json, params: { user_id: user.id }
+        expect(response.status).to eq(404)
+        error_response = JSON.parse(response.body)
+        expect(error_response["error"]).to eq("Projects not found")
+      end
+    end
+  end
+
+  describe "GET /users/:user_id/projects/:id" do
+    context "when user's project exists" do
+      it "returns the user's project" do
+        get :show, format: :json, params: { user_id: user.id, id: project_id }
+        expect(response.status).to eq(200)
+        project_response = JSON.parse(response.body)
+        expect(project_response["name"]).to eq(projects.last.name)
+        expect(project_response["description"]).to eq(projects.last.description)
+        expect(project_response["image_url"]).to eq(projects.last.image_url)
+        expect(project_response["start_date"]).to eq(projects.last.start_date)
+        expect(project_response["end_date"]).to eq(projects.last.end_date)
+        expect(project_response["duration"]).to eq(projects.last.duration)
+        expect(project_response["progress"]).to eq(projects.last.progess)
+        expect(project_response["skills"]).to eq(projects.last.skills)
+      end
+    end
+
+    context "when project does not exist" do
+      it "returns a 404 for missing project" do
+        get :show, format: :json, params: { user_id: user.id, id: 0 }
+        expect(response.status).to eq(404)
+        error_response = JSON.parse(response.body)
+        expect(error_response["error"]).to eq("Project not found")
+      end
+    end
+  end
+
+  describe "POST /users/:user_id/projects" do
+    let(:project_params) { attributes_for(:project) }
+    let(:invalid_project_params) { attributes_for(:project).merge(name: nil) }
```

Comment:
Would need to include validation for name as true in Project model, so that HTTP request will be 422 when passing in invalid params; now there will be no difference between valid & invalid